tests/Opw.PineBlog.Core.Tests/Blogs/GetBlogSettingsQueryTests.cs

```diff
@@ -34,26 +33,27 @@ public async Task Handler_Should_ReturnSettingsFromConfig_WhenNotFound()
         [Fact]
         public async Task Handler_Should_ReturnBlogSettings()
         {
+            await SeedDatabase();
+
             var result = await Mediator.Send(new GetBlogSettigsQuery());
 
             result.IsSuccess.Should().BeTrue();
             result.Value.Should().NotBeNull();
             result.Value.Title.Should().Be("blog title");
         }
 
-        private void SeedDatabase()
+        private async Task SeedDatabase()
         {
-            var context = ServiceProvider.GetRequiredService<IBlogEntityDbContext>();
+            var repo = ServiceProvider.GetRequiredService<IRepository>();
```

Comment:
Use the DbContext to seed the database, the repository is an abstraction layer. The method can then also be sync again and be called in the constructor. The tests are bound to EF since they are using the inmemory database, when we have other datasources we'll have to add tests for those specific things then.

Reply:
I was hoping to reuse those tests without modification, only swapping out the underlying data access. That is why I did not go through the DbContext.

Reply:
That would be nice, but for a future MongoDb implementation that is not possible anyway. Those tests need to have a different setup for their data. The inmemory database is also not a real database, there are differences. Complex queries should always be tested against an actual data source. But for unit testing it is sufficient.